limits.c

```diff
@@ -138,15 +138,14 @@ void limits_go_home(uint8_t cycle_mask)
   max_travel *= HOMING_AXIS_SEARCH_SCALAR; // Ensure homing switches engaged by over-estimating max travel.
   max_travel += settings.homing_pulloff;
   homing_rate = min_seek_rate * sqrt(n_active_axis); //Adjust so individual axes all move at homing rate.
-
   plan_reset(); // Reset planner buffer to zero planner current position and to clear previous motions.
 
   do {
     // Set target location and rate for active axes.
     // and reset homing axis locks based on cycle mask.
 
-    // limit travel away from the switch to the pulloff distance.
-    float travel = approach ? max_travel : settings.homing_pulloff;
+    // limit travel distance to the length of the flag
+    float travel = approach ? max_travel : MAXFLAGLEN;
```

Comment:
When homing from a state where the flag does not cross the beam, the motor travels towards the beam until the flag crosses the beam. Once this happens, it pulls off by the amount specified by `settings.homing_pulloff`. Changing `settings.homing_pulloff` to 'MAXFLAGLEN' would have drastic changes since the final homing position would then be significantly different. The `MAXFLAGLEN` value is used specifically whenever the flag is crossing the beam. It specifies how far the motor is allowed to move in order for the flag to not cross the beam. The homing bug was occurring when the flag was crossing the beam and was not allowed enough distance to travel get off the beam.

On another note, a '.json' file in 'kiosk/motion/configure' overwrites 'settings.homing_pulloff', thus changing 'DEFAULT_HOMING_PULLOFF' would actually have no effect.